sources/api/apiclient/src/exec/terminal.rs

```diff
@@ -0,0 +1,110 @@
+//! The 'terminal' module provides a Terminal type that acts as a guard around changes to terminal
+//! settings, resetting them to their original state when the Terminal is dropped.
+
+use super::get_winsize;
+use libc::{STDIN_FILENO, STDOUT_FILENO};
+use log::{debug, warn};
+use model::exec::TtyInit;
+use nix::{
+    sys::termios::{cfmakeraw, tcgetattr, tcsetattr, SetArg, Termios},
+    unistd::isatty,
+};
+use snafu::ResultExt;
+
+/// The Terminal type acts as a guard around changes to terminal settings, resetting them to their
+/// original state when the Terminal is dropped.
```

Comment:
nit: this comment seems redundant with the overall module comment

Reply:
Yeah, I was going for "useful no matter where you're looking" and didn't know how else to describe the module...